cognite/client/_api/raw.py

```diff
@@ -418,16 +423,24 @@ def _list_generator_concurrent(
             for initial in cursors
         ]
 
-        def exhaust(iterator: Iterator, results: deque[RowList], quit_early: threading.Event) -> None:
+        def exhaust(iterator: Iterator) -> None:
             for res in iterator:
                 results.append(res)
                 if quit_early.is_set():
                     return
+                # User code might be processing slower than the rate of row-fetching, and we want this
+                # iteration-based method to have a upper-bounded memory impact, so we keep a max queue
+                # size of unprocessed row-chunks:
+                while len(results) >= partitions:
```

Comment:
I have a feature request here: Can we add `or self.user_throttler()` where user_throttler is a function user can provide and control the throttling behavior of their own will? 

My user story: As a user I want to ensure the memory footprint of my data-heavy app, so I want to write the following code:
```python
import os, psutil
process = psutil.Process()

def my_throttling_logic() -> bool:
  # returns true when throttling is needed
  return process.memory_info().rss) > 2 * math.pow(1024, 3)

....
# somewhere in my code:
for rows in client.raw.rows(
            **rows_args,
            chunk_size=10_000,
            limit=-1,
            min_last_updated_time=int(_min_time),
            max_last_updated_time=int(_max_time),
            partitions=3,
            throttle=my_throttling_logic,
        )
```